### PR TITLE
safeeyes: fix double wrap

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes
 It doesn't work for me. Maybe due to the double wrap. I also remove the xprintidle-ng replacement since it's inactive while xprintidle is still active. I add wlrctl and alsa-utils dor the plugins. Some icons are missing but I don't know why. The origin fix doesn't work so I remove it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/safeeyes/default.nix [hunk 1]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
...
 
-    sed -i \
-      -e 's!xprintidle!xprintidle-ng!g' \
-      safeeyes/plugins/smartpause/config.json
+  binpath = lib.makeBinPath [
+    alsa-utils
+    wlrctl
+    xprintidle
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:57]:
```suggestion
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 2]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
...
+  postInstall = ''
+    mkdir -p $out/share/applications
+    ln -s safeeyes/platform/icons $out/share/icons
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:60]:
```suggestion
    cp safeeyes/platform/icons $out/share/icons/
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 3]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
...
+    ln -s safeeyes/platform/safeeyes.desktop $out/share/applications/safeeyes.desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:61]:
```suggestion
    cp safeeyes/platform/safeeyes.desktop $out/share/applications/safeeyes.desktop
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 4]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
...
-    mkdir -p $out/share/applications
-    cp -r safeeyes/platform/icons $out/share/
-    cp safeeyes/platform/safeeyes.desktop $out/share/applications/
   '';
 
   doCheck = false; # no tests
 
-  meta = {
+  # handle setup hooks better
+  strictDeps = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:74]:
```suggestion
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 5]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
   buildInputs = [
     gtk3
     gobject-introspection
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:31]: This must go to nativeBuildInputs for the setup hook to work.

File: pkgs/applications/misc/safeeyes/default.nix [hunk 6]
```diff
@@ -16,59 +29,55 @@ in buildPythonApplication rec {
...
-    gappsWrapperArgs+=(
-      --prefix XDG_DATA_DIRS : "${gdk-pixbuf}/share"
-      --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
-      --prefix XDG_DATA_DIRS : "${librsvg}/share"
-
-      # safeeyes images
-      --prefix XDG_DATA_DIRS : "$out/lib/${python.libPrefix}/site-packages/usr/share"
+    makeWrapperArgs+=(
+      "''${gappsWrapperArgs[@]}"
+      --prefix PATH : ${binpath}
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:67]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ alsa-utils wlrctl xprintidle ];
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/safeeyes/default.nix (hunk 5 above)
Comment (linsui) [reply pkgs/applications/misc/safeeyes/default.nix:31]: I tried that but there is no difference.

File: pkgs/applications/misc/safeeyes/default.nix [hunk 7]
```diff
@@ -16,59 +29,47 @@ in buildPythonApplication rec {
   buildInputs = [
     gtk3
     gobject-introspection
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:31]:
```suggestion
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 8]
```diff
@@ -16,59 +29,47 @@ in buildPythonApplication rec {
...
-    gnome.adwaita-icon-theme
-    gnome.adwaita-icon-theme
+    libappindicator-gtk3
+    libnotify
   ];
 
   nativeBuildInputs = [
     wrapGAppsHook
+    gobject-introspection
   ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:40]:
```suggestion
```

File: pkgs/applications/misc/safeeyes/default.nix [hunk 9]
```diff
@@ -16,59 +29,47 @@ in buildPythonApplication rec {
   buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/safeeyes/default.nix:29]:
```suggestion
  nativeBuildInputs = [
    wrapGAppsHook
    gobject-introspection
  ];

  buildInputs = [
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

**Comment by linsui** [comment]
Thanks!

**Comment by fgaz** [comment]
Too bad... I guess it can wait until 22.11

**Comment by linsui** [comment]
You can backport it manully anyway.

**Comment by fgaz** [comment]
Too lazy :-P

**Pull request merged**